pkg/cloudprovider/vsphere/cloud.go

```diff
@@ -113,6 +127,17 @@ func (vs *VSphere) Initialize(clientBuilder cloudprovider.ControllerClientBuilde
 		vs.connectionManager = connMgr
 		vs.nodeManager.connectionManager = connMgr
 
+		logoutWG.Add(1)
+		// Gracefully logout of all VSphere sessions if the stop channel is signaled
+		go func() {
+			// TODO: The stop channel is unusable since the cloud-provider module
```

Comment:
Could you add it's because of this line https://github.com/kubernetes/cloud-provider/blob/1bae60eb89ced16795f81a900b79cb55524ba6f0/app/controllermanager.go#L536
Not using `ctx := wait.ContextForChannel(stopCh)` I assume?
Also would be useful to open an issue in cloud-provider repo.
Good catch!

Reply:
Added a comment to point to the `context.TODO()`.